internal/info/info.go

```diff
@@ -247,7 +247,7 @@ func (d Detail) String() string {
 // Get returns parased Detail object.
 func (i info) Get() Detail {
 	i.prepare()
-	defaultURL := fmt.Sprintf("https://%s/tree/%s", valdRepo, i.detail.GitCommit)
+	defaultURL := "https://" + valdRepo + "/tree/" + i.detail.GitCommit
```

Comment:
I think we can initialize it when we initialize `info`.

Reply:
um... `defaultURL` in this function may not be used.
we can add default case to initialize url here https://github.com/vdaas/vald/blob/e4410016fa8201ec27f47ae5da10fea979298d52/internal/info/info.go#L263-L282

```
switch {
		case strings.HasPrefix(file, runtime.GOROOT()+"/src"):
		......
		case strings.Contains(file, "go/src/") && strings.Contains(file, valdRepo):
			url = strings.Replace(strings.SplitN(file, "go/src/", 2)[1]+"#L"+strconv.Itoa(line), valdRepo, "https://"+valdRepo+"/blob/"+i.detail.GitCommit, -1)
                 default:
                          url = "https://" + valdRepo + "/tree/" + i.detail.GitCommit
		}
```

Reply:
I don't think a string can be initialized when info is initialized.
For example, how about doing it in the `prepare` function?

Reply:
how about we initialize it at prepare function and hold them at struct field ?
and runtime.GOROOT as well.

Reply:
runtime.GOROOT() also have overhead to fetch environment variable.
FYI: https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/runtime/extern.go;l=243

Reply:
Thank you for your comment.
I have fixed: https://github.com/vdaas/vald/pull/1617/commits/12b8de8831532bebcf52f0e1abfb34379e139dd5

Reply:
@kpango 
Thank you for sharing. I will fix it soon!

Reply:
I have fixed based on two comment 👍 

https://github.com/vdaas/vald/pull/1617/commits/12b8de8831532bebcf52f0e1abfb34379e139dd5
https://github.com/vdaas/vald/pull/1617/commits/3b4ae955be33b8577886c067efedad5aae7d057b

Reply:
`prepare()` is called every time when `Get()` is called, so it may not effective if we implement it in `prepare()`.
I think `goRoot` can be initialize when we start using this package.

Reply:
can you please read code？